File: RecorridoBinario.js

function findEle(arr, l, h, targetValue) {
    while (l <= h) {
        
        let mid = Math.floor(l + (h - l) / 2);

        if (arr[mid] === targetValue) {
            return mid;
        }
        
        else if (arr[mid] < targetValue) {
            l = mid + 1;
        }
      
        else {
            h = mid - 1;
        }
    }
    return -1; 
}


const inputArr = [1, 2, 3, 4, 5, 6, 7, 8, 9, 0]; 
const targetElement = 3;
const s = inputArr.length;

const idx = findEle(inputArr, 0, s - 1, targetElement);

if (idx !== -1) {
    console.log("El elemento se encuentra en la posición: " + (idx + 1));
} else {
    console.log("El elemento no se encuentra.");
}

File: RecorridoBinario.py
def findEle(arr, l, h, targetValue):
    while l <= h:
        mid = l + (h-l) // 2

        if arr[mid] == targetValue:
            return mid
        elif arr[mid] < targetValue:
            l = mid + 1
        else:
            h = mid - 1
    return -1

if __name__ == '__main__':
    inputArr = [1,2,3,4,5,6,7,8,9,0]
    targetElement = 6
    s = len(inputArr)

    idx = findEle(inputArr, 0, s - 1, targetElement)

    if idx != -1:
        print("El elemento se encuentra en la posicion: " + str(idx + 1))
    else:
        print("El elemento no se encuentra")